lib/src/values/simple_config_object.cc

```diff
@@ -390,4 +400,104 @@ namespace hocon {
         }
     }
 
+    bool compare(string& a, string& b) {
```

Comment:
http://en.cppreference.com/w/cpp/algorithm/sort notes that this function should use `string const&` arguments. Specifically, the signature is `bool cmp(const Type1 &a, const Type2 &b);`. That's probably what's causing the compile error.